### PR TITLE
Add post on resources for learning Go

### PR DESCRIPTION
**Pull request by nickcharlton**

### CODE REVIEW

File: _posts/2019-10-20-learning-go-resources.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+---
+title: "Some resources for learning Go"
+published: 2019-10-20 13-01-02 +01:00
+tags: programming go
+---
+
+I'm starting on a Go project for the first time in a while, and a few people
+have asked me how to starting writing Go. The last time I spent a significant
+time writing Go was over a year ago, so while I also needed to refresh my
```
Comment (rossbearman) [inline _posts/2019-10-20-learning-go-resources.md]: "a significant **amount of** time"

File: _posts/2019-10-20-learning-go-resources.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+knowledge, there have been a few new things released since.
+
+I learn best in two phases: first by reading a lot about a subject, and then
+building something. The project itself is the last bit, but going into it, I
```
Comment (rossbearman) [inline _posts/2019-10-20-learning-go-resources.md]: > first ~~by~~ reading a lot about a subject **[dropped comma]** and then building something **with it**.

File: _posts/2019-10-20-learning-go-resources.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+want to be as primed as I can to hit the ground running. That all said, _what_
```
Comment (rossbearman) [inline _posts/2019-10-20-learning-go-resources.md]: > but going into it **[dropped comma]** I want to be as **ready** as I can to hit the ground running

File: _posts/2019-10-20-learning-go-resources.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+I read is essential. I don't find programming books which walk through things
+like type primitives in great depth remotely engaging &mdash; I want to know
+just enough to be productive.
+
+I'm assuming here that you're coming from the position of already being
```
Comment (rossbearman) [inline _posts/2019-10-20-learning-go-resources.md]: > I'm assuming ~~here~~ that you're coming

File: _posts/2019-10-20-learning-go-resources.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+## [Go in Practice](http://goinpracticebook.com)
+
+Once you've got an understanding of the basics of Go and have written a little,
+I'd recommend _Go in Practice_. It'll run through some common things you'll
+need to do and expose you to some patterns as you go. You might well find you
+can use the examples mostly as-is, which I find helps in making it feel like
+you're making good progress in building something.
+
+I read this cover to cover, but you don't need to do that. I find once I have,
+though, it's much easier for me to pick the book up and refer to something I
```
Comment (rossbearman) [inline _posts/2019-10-20-learning-go-resources.md]: >I find once I have ~~, though,~~ it's much easier

File: _posts/2019-10-20-learning-go-resources.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+then need.
+
+## [Ultimate Go Programming Course]()
+
+
+Once I've built a few things, I then want to go into some more depth. I find
```
Comment (rossbearman) [inline _posts/2019-10-20-learning-go-resources.md]: >Once I've built a few **simple projects** ~~things~~, I then want to go into some more depth **[explain]**.

e.g.

>I then want to go into some more depth **on how to structure more complex applications**

File: _posts/2019-10-20-learning-go-resources.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+this sort of depth doesn't help until much later on and watching a video course
+helps keep me engaged.
+
+I've been finding Bill Kennedy's _Ultimate Go Programming_ fit this well, he's
+an engaging teacher, and the depth is just right.
```
Comment (rossbearman) [inline _posts/2019-10-20-learning-go-resources.md]: >an engaging teacher **[dropped comma]** and the depth is just right.

File: _posts/2019-10-20-learning-go-resources.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+[Ultimate Go Programming Course]: https://itp.pearson.com/courses/course-v1:ITP+9780135261637+2019/about
+[oreilly]: https://learning.oreilly.com/videos/ultimate-go-programming/9780134757476
+
+## [Go by Example](https://gobyexample.com)
+
+Finally, something I refer to regularly that'll help fill in some gaps. I jump
+to _Go by Example_ when I broadly know what I'm trying to do, but can't quite
+figure out how to do it in Go.
+
+It's how I solve something like: "right I need to read a file …how do I do
```
Comment (rossbearman) [inline _posts/2019-10-20-learning-go-resources.md]: >"~~right~~ I need to read a file …how do I do that?"

File: _posts/2019-10-20-learning-go-resources.md (hunk 6 above)
Comment (nickcharlton) [reply _posts/2019-10-20-learning-go-resources.md]: Nice. I think this is something I'm likely to try and write up next. Maybe a tour of how to structure Go projects.

**Pull request merged**